Relevant history of the `SDL2` package and the `moveToOutput` in `postInstall`:

```
commit b81caba5fbb49012a5cabefb672f3b83bd524d1e
Author: Carles Pagès <page@cubata.homelinux.net>
Date:   Wed Nov 6 23:17:46 2013 +0100

   SDL2: some improvements to the expression.

commit da70d3da0f11b22eac77756b39b349215e06b2e3
Author: Linus Heckemann <git@sphalerite.org>
Date:   Sun Dec 25 01:11:07 2016 +0000

   SDL2: split derivation

commit 0c42efd9d70774eafb47212967caada630335731
Author: Lluís Batlle i Rossell <viric@viric.name>
Date:   Thu Feb 16 22:24:40 2017 +0100

   SDL2: fix creation of libSDL2main.a
   
   It's required by a trigger rally update I will commit next.
   And other games use that too.

commit 1e7da9ec78d11f586fc74747d77b17f91969a7ed
Author: Lluís Batlle i Rossell <viric@viric.name>
Date:   Thu Feb 16 22:25:23 2017 +0100

   trigger: update to 0.6.5
```

What the `libtool` problem looks like when static libs are enabled (this PR fixes that):

```
libtool: link: gcc -shared  .libs/SDL_ttf.o   -Wl,-rpath -Wl,/nix/store/ggafj30pr2sv0hj7jgfywvqv1yz88mws-freetype-2.10.1/lib -Wl,-rpath -Wl,/nix/store/mdxphvay0gaxrrm7hpanfc76j198bbvj-bzip2-1.0.6.0.1/lib -Wl,-rpath -Wl,/nix/store/q33gb957w94mqdqkq4pwil8hix34krwv-libpng-apng-1.6.37/lib -Wl,-rpath -Wl,/nix/store/x3szqg2c1rav8ja944583r1a1m6vmq86-SDL2-2.0.10/lib -Wl,-rpath -Wl,/nix/store/ggafj30pr2sv0hj7jgfywvqv1yz88mws-freetype-2.10.1/lib -Wl,-rpath -Wl,/nix/store/mdxphvay0gaxrrm7hpanfc76j198bbvj-bzip2-1.0.6.0.1/lib -Wl,-rpath -Wl,/nix/store/q33gb957w94mqdqkq4pwil8hix34krwv-libpng-apng-1.6.37/lib -Wl,-rpath -Wl,/nix/store/x3szqg2c1rav8ja944583r1a1m6vmq86-SDL2-2.0.10/lib -L/nix/store/ggafj30pr2sv0hj7jgfywvqv1yz88mws-freetype-2.10.1/lib /nix/store/ggafj30pr2sv0hj7jgfywvqv1yz88mws-freetype-2.10.1/lib/libfreetype.so -L/nix/store/2fd7k3j5vmmkzw14ywwjykw8sa4h0vdd-zlib-1.2.11/lib -L/nix/store/mdxphvay0gaxrrm7hpanfc76j198bbvj-bzip2-1.0.6.0.1/lib -L/nix/store/q33gb957w94mqdqkq4pwil8hix34krwv-libpng-apng-1.6.37/lib /nix/store/mdxphvay0gaxrrm7hpanfc76j198bbvj-bzip2-1.0.6.0.1/lib/libbz2.so /nix/store/q33gb957w94mqdqkq4pwil8hix34krwv-libpng-apng-1.6.37/lib/libpng16.so -lz -L/nix/store/x3szqg2c1rav8ja944583r1a1m6vmq86-SDL2-2.0.10/lib /nix/store/x3szqg2c1rav8ja944583r1a1m6vmq86-SDL2-2.0.10/lib/libSDL2.so -lm -lX11 -lXext -lXcursor -lXinerama -lXi -lXrandr -lXss -lXxf86vm -lpthread -lrt  -Wl,-rpath -Wl,/nix/store/x3szqg2c1rav8ja944583r1a1m6vmq86-SDL2-2.0.10/lib -Wl,--enable-new-dtags   -Wl,-soname -Wl,libSDL2_ttf-2.0.so.0 -o .libs/libSDL2_ttf-2.0.so.0.14.1
checking if g++ static flag -static works... yes
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXext
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXcursor
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXinerama
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXi
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXrandr
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXss
/nix/store/f1v9hcm7pxdrgmmarf5ldss9xc480l0n-binutils-2.31.1/bin/ld: cannot find -lXxf86vm
collect2: error: ld returned 1 exit status
make: *** [Makefile:517: libSDL2_ttf.la] Error 1

builder for '/nix/store/x1gslck2ab1r09by2jgpm6qzcn24d90a-SDL2_ttf-2.0.15.drv' failed with exit code 2
```

--- a/pkgs/development/libraries/SDL2/default.nix
+++ b/pkgs/development/libraries/SDL2/default.nix
@@ -75,9 +75,21 @@ stdenv.mkDerivation rec {
     ++ optional alsaSupport "--with-alsa-prefix=${alsaLib.out}/lib"
     ++ optional stdenv.isDarwin "--disable-sdltest";
 
+  # We remove libtool .la files when static libs are requested,
+  # because they make the builds of downstream libs like `SDL_tff`
+  # fail with `cannot find -lXext, `-lXcursor` etc. linker errors
+  # because the `.la` files are not pruned if static libs exist
+  # (see https://github.com/NixOS/nixpkgs/commit/fd97db43bcb05e37f6bb77f363f1e1e239d9de53)
+  # and they also don't carry the necessary `-L` paths of their
+  # X11 dependencies.
+  # For static linking, it is better to rely on `pkg-config` `.pc`
+  # files.
   postInstall = ''
-    moveToOutput lib/libSDL2main.a "$dev"
-    rm $out/lib/*.a
+    if [ "$dontDisableStatic" -eq "1" ]; then
+      rm $out/lib/*.la
+    else
+      rm $out/lib/*.a
+    fi
     moveToOutput bin/sdl2-config "$dev"
   '';
 

